csrc/fusion.h

```diff
@@ -348,8 +348,51 @@ class TORCH_CUDA_CU_API Fusion : public IrContainer {
     return std::any_cast<T&>(managed_named_data_.at(key).first);
   }
 
+  //! Try to get managed data by index, checking that we have an entry for it,
+  //! and that the entry has not been reset (see stopManaging).
+  template <typename T>
+  inline std::optional<const T> getManagedSafe(size_t index) const {
+    if (hasManaged(index)) {
+      return std::any_cast<T>(managed_data_.at(index).first);
+    }
+    return std::nullopt;
+  }
+
+  //! Try to get managed data by key, checking that we have an entry for that
+  //! key.
+  template <typename T>
+  inline std::optional<const T> getManagedSafe(std::string key) const {
```

Comment:
@zasdfgbnm I had a little trouble trying to return an optional non-const reference, so I just added a const "safe" version here. The existing interface with the unchecked ("unsafe") versions is unmodified.